front_end/content.py

```diff
@@ -262,6 +264,18 @@ def administrator_exists(self):
     def is_administrator(self, user_id):
         return self.user_has_role(user_id, 0, "administrator")
 
+    def is_student(self, user_id):
```

Comment:
Although this logic would work, it would mean that we would hit the database one more time for every time that a page is loaded. Also, this function should be course-specific. A user could be an instructor (or assistant) in one course but a student in another course.

Unless I'm missing something, we should be able to remove this function from content.py and replace it with a function in BaseUserHandler in webserver.py. That function should look something like this:

```
def is_student_for_course(self, course_id):
    return not self.is_administrator() and not self.is_instructor_for_course(course_id) and not self.is_assistant_for_course(course_id)
```

Those functions use context variables, which will be fast.